src/alarms/notifications/NewNotification.js

```diff
@@ -81,20 +78,62 @@ async function fetchNestedAssets(assetType, assetId) {
     )
       .then(response => response.json())
       .then(data => {
-        console.log("NewNotification fetchNestedAssets data", data);
-        console.log(
-          "NewNotification fetchNestedAssets data.filters",
-          data.filters
-        );
-        return data["filters"];
+        let nestedAssetsList = [];
+        if (data.filters) {
+          nestedAssetsList = data.filters;
+        } else if (data.pumps) {
+          nestedAssetsList = data.pumps;
+        }
+        return nestedAssetsList;
       });
-    console.log("NewNotification fetchNestedAssets nestedAssets", nestedAssets);
     return nestedAssets;
   } catch (e) {
     throw new Error(e);
   }
 }
 
+async function fetchTimeseriesUuidsFromAsset(assetType, assetId) {
+  // Get timeserie uuids from the asset and its nested assets.
+  try {
+    // Set page_size to 100000, same as in Raster.js
+    const uuids = await fetch(
+      `/api/v3/${assetType}s/${assetId}/?page_size=100000`,
+      {
+        credentials: "same-origin"
+      }
+    )
+      .then(response => response.json())
+      .then(data => {
+        let timeseriesUuid = [];
+        // Check if the asset has timeseries.
+        if (data.timeseries) {
+          timeseriesUuid = data.timeseries;
+        }
+        // Check if the asset has filters with timeseries.
+        if (data.filters && data.filters[0] && data.filters[0].timeseries) {
+          let dataFilterUuids = data.filters.map(function(dataFilter) {
+            dataFilter.timeseries.forEach(function(dataFilterTimeserie) {
+              timeseriesUuid.push(dataFilterTimeserie.uuid);
+            });
+          });
+          timeseriesUuid.concat(dataFilterUuids);
+        }
+        // Check if the asset has pumps with timeseries.
+        if (data.pumps && data.pumps[0] && data.pumps[0].timeseries) {
+          let dataPumpUuids = data.pumps.map(function(dataPump) {
+            dataPump.timeseries.forEach(function(dataPumpTimeserie) {
+              timeseriesUuid.push(dataPumpTimeserie.uuid);
+            });
+          });
+        }
```

Comment:
Omdat dit twee keer hetzelfde is kun je het ook afsplitsen en loopen over `['pumps', 'filters']`. Iets als `['pumps', 'filters'].forEach((nestedAsset) => { if data[nestedAsset] && data[nestedAsset][0]... etc`.